### PR TITLE
Change empty matric display to display nothing

### LINKED ISSUE #181: Hide Empty Matriculation Numbers
We want to avoid showing the matriculation numbers so that each contact entry looks complete.

### PR DESCRIPTION
**Pull request by jayllo-c**
Fixes #181 

Currently, when the matriculation value is empty, the text
"No matriculation number" is displayed in the person card in the UI.

However, this may cause the user to perceive this as an issue rather
than a feature since this list may contain staff who do not have
matriculation numbers.

Let's hide the matriculation number if it is empty.

As a result, each entry will only show the information available, which
makes reading data from the app more concise.

### CODE REVIEW

File: src/main/java/seedu/address/model/student/Matric.java [hunk 1]
```diff
@@ -74,7 +74,6 @@ public int hashCode() {
      * @return String representation of the matriculation number
      */
     public String toString() {
-        return isEmptyMatric(matricNumber) ? "No matriculation number" : matricNumber;
+        return isEmptyMatric(matricNumber) ? "" : matricNumber;
```
Comment (Zer0Legion) [inline src/main/java/seedu/address/model/student/Matric.java:77]: Great refactor! Now in v1.3 we do not need such simplistic and explicit messages that a field was left empty.

**Review by delishad21 — COMMENTED** [review]
Good change. The UI looks a lot cleaner without a placeholder indicator for empty matric numbers.

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 2]
```diff
@@ -59,6 +59,10 @@ public PersonCard(Person person, int displayedIndex) {
         address.setText(person.getAddress().value);
         email.setText(person.getEmail().value);
         matric.setText(person.getMatric().toString());
+        if (matric.getText().isEmpty()) {
```
Comment (Pughal77) [inline src/main/java/seedu/address/ui/PersonCard.java:62]: Nice you have edited the code such that it checks if the matric field is empty without causing any bugs for setting the text for other fields

**Review by Pughal77 — COMMENTED** [review]
Good, clean addition to our program!

**Pull request merged**